plugins/cameraserver/src/main/java/edu/wpi/first/shuffleboard/plugin/cameraserver/source/CameraUrlGenerator.java

```diff
@@ -45,31 +51,95 @@ public CameraUrlGenerator(CameraServerSource source) {
     if (frameRate > 0) {
       commands.put("fps", Integer.toString(frameRate));
     }
-    return generateUrls(commands, baseUrls);
+    // Add a special case command for NI Cameras
+    if (source.getName().contains("IMAQdx")) {
+      commands.put("name", source.getName());
+    }
+    return generateUrls(commands, baseUrls, source.getName());
   }
 
   @VisibleForTesting
-  static String[] generateUrls(Map<String, String> commands, String[] baseUrls) { // NOPMD varargs instead of array
+  static String[] generateUrls(Map<String, String> commands, String[] baseUrls,
+                               String cameraName) {
```

Comment:
The `cameraName` parameter is no longer used in the method body and should be removed. The NOPMD comment doesn't need to be re-added, since we've updated our PMD ruleset since this method was originally written.

Reply:
Fixed